R/estimateBaseline.R

```diff
@@ -0,0 +1,242 @@
+##'   determining the half size of the moving window for the TopHat
+##'   filter.
+##'
+##' - ConvexHull: The baseline estimation is based on a convex hull
+##'   constructed below the spectrum.
+##'
+##' - Median: This baseline estimation uses a moving median. It is
+##'   based on [stats::runmed()]. The additional argument
+##'   `halfWindowSize` corresponds to the `k` argument in
+##'   [stats::runmed()] (`k = 2 * halfWindowSize + 1`) and controls
+##'   the half size of the moving window.
+##'
+##' @return `numeric()` with estimated baseline intensities.
+##'
+##' @references
+##'
+##' These functions have been ported from the
+##' [MALDIqaunt](https://strimmerlab.github.io/software/maldiquant/)
+##' package.
+##'
+##' SNIP:
+##'
+##' - C.G. Ryan, E. Clayton, W.L. Griffin, S.H. Sie, and D.R. Cousens
+##'   (1988). Snip, a statistics-sensitive background treatment for
+##'   the quantitative analysis of pixe spectra in geoscience
+##'   applications. Nuclear Instruments and Methods in Physics
+##'   Research Section B: Beam Interactions with Materials and Atoms,
+##'   34(3): 396-402.
+##'
+##' - M. Morhac (2009). An algorithm for determination of peak regions
+##'   and baseline elimination in spectroscopic data. Nuclear
+##'   Instruments and Methods in Physics Research Section A:
+##'   Accelerators, Spectrometers, Detectors and Associated Equipment,
+##'   600(2), 478-487.
+##'
+##' TopHat:
+##'
+##' - M. van Herk (1992). A Fast Algorithm for Local Minimum and
+##'   Maximum Filters on Rectangular and Octagonal Kernels. Pattern
+##'   Recognition Letters 13.7: 517-521.
+##'
+##' - J. Y. Gil and M. Werman (1996). Computing 2-Dimensional Min,
+##'   Median and Max Filters. IEEE Transactions: 504-507.
+##'
+##' ConvexHull:
+##'
+##' - Andrew, A. M. (1979). Another efficient algorithm for convex
+##'   hulls in two dimensions. Information Processing Letters, 9(5),
+##'   216-219.
+##'
+##' @author Sebastian Gibb
+##'
+##' @importFrom stats runmed
+##'
+##' @useDynLib MsCoreUtils, .registration = TRUE
+##'
+##' @export
+##'
+##' @examples
+##'
+##' ## ----------------------------
+##' ## Simulation example data
+##' nmz <- 5000
+##' mz <- seq(1000, length.out = nmz)
+##' ## create peaks
+##' center <- seq(50, nmz, by = 500)
+##' peaks <- lapply(center, function(cc)1000 * dpois(0:100, (1000 + cc) / 75))
+##' ## create baseline
+##' intensity <- 100 * exp(-seq_len(nmz)/2000)
+##' ## add peaks to baseline
+##' for (i in seq(along = center)) {
+##' intensity[center[i]:(center[i] + 100)] <-
+##'    intensity[center[i]:(center[i] + 100)] + peaks[[i]]
+##' }
+##' ## add noise
+##' intensity <- intensity + rnorm(nmz, mean = 0, sd = 1)
+##'
+##' plot(mz, intensity, type = "l")
+##'
+##' ## ----------------------------
+##' ## SNIP baseline
+##' base_SNIP <- estimateBaseline(mz, intensity,
+##'                               method = "SNIP",
+##'                               iterations = 100L)
```

Comment:
```suggestion
##'                               iterations = 20L)
```

I just adapted the iterations/hws to fit better to the example data.